api/src/main/java/net/kyori/adventure/animation/Animation.java

```diff
@@ -0,0 +1,276 @@
+   *
+   * @since 4.9.2
+   */
+  static TitleAnimation title(int titleOffset, ComponentAnimation titleAnimation, ComponentAnimation subtitleAnimation, Title.Times times) {
+    final List<Title> frames = new ArrayList<>();
+
+    final Iterator<Component> titleIterator = titleAnimation.frameIterator();
+    final Iterator<Component> subtitleIterator = subtitleAnimation.frameIterator();
+
+    while (titleIterator.hasNext() || titleOffset > 0 || subtitleIterator.hasNext()) {
+      final Component subtitle;
+
+      if (subtitleIterator.hasNext())
+        subtitle = subtitleIterator.next();
+      else
+        subtitle = Component.empty();
+
+      final Component title;
+
+      if (titleOffset <= 0) {
+        if (titleIterator.hasNext())
+          title = titleIterator.next();
+        else
+          title = Component.empty();
+      } else {
+        title = Component.empty();
+        titleOffset--;
+      }
+
+      frames.add(Title.title(title, subtitle, times));
+    }
+
+    return new TitleAnimationImpl(frames);
+  }
+
+  /**
+   * Does this same as @link{TitleAnimation#title(int, ComponentAnimation, ComponentAnimation, Title.Times)}, but with default times.
+   *
+   * @param titleOffset an offset of title frames
+   * @param titleAnimation an animation representing title part
+   * @param subtitleAnimation an animation representing subtitle part
+   * @return new title animation from parameters
+   * @since 4.9.2
+   */
+  static TitleAnimation title(int titleOffset, ComponentAnimation titleAnimation, ComponentAnimation subtitleAnimation) {
+    return Animation.title(titleOffset, titleAnimation, subtitleAnimation, Title.DEFAULT_TIMES);
+  }
+
+  /**
+   * Creates a title animation of two component animations representing frame streams for title and subtitle.
+   *
+   * <p>Both title and subtitle animations start from this same point.</p>
+   *
+   * @param titleAnimation an animation representing title part
+   * @param subtitleAnimation an animation representing subtitle part
+   * @param times times for all frames
+   * @return new title animation from parameters
+   *
+   * @since 4.9.2
+   */
+  static TitleAnimation title(ComponentAnimation titleAnimation, ComponentAnimation subtitleAnimation, Title.Times times) {
+    return Animation.title(titleAnimation, 0, subtitleAnimation, times);
+  }
+
+  /**
+   * Does this same as @link{TitleAnimation#title(ComponentAnimation, ComponentAnimation, Title.Times)}, but with default times.
+   *
+   * @param titleAnimation an animation representing title part
+   * @param subtitleAnimation an animation representing subtitle part
+   * @return new title animation from parameters
+   * @since 4.9.2
+   */
+  static TitleAnimation title(ComponentAnimation titleAnimation, ComponentAnimation subtitleAnimation) {
+    return Animation.title(titleAnimation, 0, subtitleAnimation, Title.DEFAULT_TIMES);
+  }
+
+  /**
+   * Returns a list of frames contained by this animation.
```

Comment:
Is this list mutable or immutable? Generally, mutable lists are frowned upon by the Adventure API unless it is absolutely unavoidable (such as builders or boss bar objects).